daliuge-engine/dlg/apps/dockerapp.py

```diff
@@ -350,27 +351,22 @@ def run(self):
 
         c = DockerApp._get_client()
 
-        # Remove the container unless it's specified that we should keep it
-        # (used below)
-        def rm(container):
-            if self._removeContainer:
-                container.remove()
-
         # Create container
-        container = c.containers.create(
+        self.container = c.containers.create(
                 self._image,
                 cmd,
                 volumes=binds,
                 user=user,
                 environment=env,
-                working_dir=self.workdir
+                working_dir=self.workdir,
```

Comment:
supposedly adding `init=True` here reserves PID 1 to a lightweight process for handling interupts which may improve performance of container stopping. Will test this some more.